packages/muelu/test/tutorial/ScalingTest.cpp

```diff
@@ -0,0 +1,695 @@
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact
+//                    Jonathan Hu       (jhu@sandia.gov)
+//                    Andrey Prokopenko (aprokop@sandia.gov)
+//                    Ray Tuminaro      (rstumin@sandia.gov)
+//
+// ***********************************************************************
+//
+// @HEADER
+#include <unistd.h>
+#include <iostream>
+
+// Teuchos
+#include <Teuchos_RCP.hpp>
+#include <Teuchos_ParameterList.hpp>
+#include <Teuchos_CommandLineProcessor.hpp>
+#include <Teuchos_GlobalMPISession.hpp>
+#include <Teuchos_DefaultComm.hpp>
+
+// Xpetra
+#include <Xpetra_Map.hpp>
+#include <Xpetra_MapFactory.hpp>
+#include <Xpetra_CrsMatrixWrap.hpp>
+#include <Xpetra_VectorFactory.hpp>
+#include <Xpetra_MultiVectorFactory.hpp>
+#include <Xpetra_Parameters.hpp>
+
+// Galeri
+#include <Galeri_XpetraParameters.hpp>
+#include <Galeri_XpetraProblemFactory.hpp>
+#include <Galeri_XpetraUtils.hpp>
+
+// MueLu
+#include "MueLu_ConfigDefs.hpp"
+#include "MueLu_Memory.hpp"
+#include "MueLu_Hierarchy.hpp"
+#include "MueLu_SaPFactory.hpp"
+#include "MueLu_RAPFactory.hpp"
+#include "MueLu_TrilinosSmoother.hpp"
+#include "MueLu_DirectSolver.hpp"
+#include "MueLu_Utilities.hpp"
+#include "MueLu_Exceptions.hpp"
+#include "MueLu_UncoupledAggregationFactory.hpp"
+#include "MueLu_TentativePFactory.hpp"
+#include "MueLu_TransPFactory.hpp"
+#include "MueLu_SmootherFactory.hpp"
+#include "MueLu_RepartitionFactory.hpp"
+#include "MueLu_RebalanceTransferFactory.hpp"
+#include "MueLu_CoordinatesTransferFactory.hpp"
+#include "MueLu_ZoltanInterface.hpp"
+#include "MueLu_RebalanceAcFactory.hpp"
+#include "MueLu_CoalesceDropFactory.hpp"
+
+// Belos
+#ifdef HAVE_MUELU_BELOS
+#include "BelosConfigDefs.hpp"
+#include "BelosLinearProblem.hpp"
+#include "BelosBlockCGSolMgr.hpp"
+#include "BelosBlockGmresSolMgr.hpp"
+#include "BelosXpetraAdapter.hpp" // this header defines Belos::XpetraOp()
+#include "BelosMueLuAdapter.hpp"  // this header defines Belos::MueLuOp()
+#endif
+
+#ifdef HAVE_MUELU_ISORROPIA
+#include "MueLu_IsorropiaInterface.hpp"
+#endif
```

Comment:
Zoltan2?

Reply:
@cgcgcg The tests needs some more attention. Right now, they are running on the `Epetra` stack. I'd like them to be converted to `Tpetra` and then call MueLu either with `Tpetra` or `Xpetra` objects. This will automatically modernize the stack of adjacent tools.